### ISSUE TITLE
net-plugin v2 fastdp not working for new installs or on clean reboot

### ISSUE THREAD

**Issue by dtshepherd** [weaveworks/weave#3087]
## What you expected to happen?

Weave connections would be `fastdp` on install and/or clean reboot.

## What happened?
During clean install of weave net-plugin (v2) only uses `sleeve` connections rather than `fastdp`.  It is missing links `datapath`, `vethwe-datapath`, and `vxlan-6784`.

Disabling the docker plugin and running `weave launch`, `weave reset --force`, then re-enabling the plugin _sometimes_ gets the links to switch to `fastdp`.  Once `fastdp` is working, disabling/enabling/re-installing the net-plugin is safe to do (continues to use `fastdp`).  However, rebooting the machine puts the links back into `sleeve` mode again.

## How to reproduce it?
* Spin up two clean CentOS 7 or Ubuntu 16.04 machines (I used AWS)
* On each node...
  * Install Docker
  * Swarm Init and Join the two machines
  * Install `latest_release` Weave Plugin V2 on both machines
  * Install `weave` shell script
  * Run `weave status connections` and observe only `sleeve` connections
  * Run `ifconfig` or `ip link` to see missing `datapath` links
  * Disable weave plugin
  * Try to get `datapath` to show up by running weave container manually
    * Run `weave launch`
    * Check for `datapath`, if found run `weave stop` and exit loop
    * Run `weave reset --force`
    * Repeat until `datapath` shows up
`weave stop`, until `datapath` link shows up
  * Re-enable weave plugin
  * Run `weave status connections` to confirm `fastdp` connections

## Anything else we need to know?
This happens on AWS CentOS 7.3 or Ubuntu 16.04 images.  I can't seem to consistently kick the `datapath` link, hence the terrible looping workaround.  `lsmod` is missing `openvswitch` and `vport_vxlan`, however, that may be a red herring.  I tried `modprobe` inbetween an disable/enable, which did not help.

I've also seen a _normal_ `weave launch` not create the `datapath` link, but I can't seem to consistently reproduce that issue.

I initially thought this was a kernel version issue, because `fastdp` encryption says it requires kernel 4.2, but I eliminated that concern when I could reproduce the problem on Ubuntu 16.04 with the 4.4 kernel with and without encryption.

Started a conversation in Slack about a few weeks ago as well:
https://weave-community.slack.com/archives/C2ND76PAA/p1501719958775811

## Versions:

CentOS 7:
```
[centos@ip-172-31-0-27 ~]$ weave version
weave script 2.0.1
weave 2.0.1
[centos@ip-172-31-0-27 ~]$ docker version
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:20:36 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:21:56 2017
 OS/Arch:      linux/amd64
 Experimental: false
[centos@ip-172-31-0-27 ~]$ uname -a
Linux ip-172-31-0-27.ec2.internal 3.10.0-514.16.1.el7.x86_64 #1 SMP Wed Apr 12 15:04:24 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```
Ubuntu 16.04:
```
dtshepherd@NUC:~$ weave version
weave script 2.0.1
weave 2.0.1
dtshepherd@NUC:~$ docker version
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:23:31 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:19:04 2017
 OS/Arch:      linux/amd64
 Experimental: false
dtshepherd@NUC:~$ uname -a
Linux NUC 4.4.0-87-generic #110-Ubuntu SMP Tue Jul 18 12:55:35 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
dtshepherd@NUC:~$ 
```

## Logs:

I captured my shell commands and good/bad docker log snippets:
https://gist.github.com/dtshepherd/e95975d6667e07d92354458d35e7d48c

Nothing stands out in the logs as to why it couldn't setup `fastdp`.  I only see a log message saying `Sniffing traffic on vethwe-pcap (via pcap)` (when it uses `sleeve`) or `Sniffing traffic on datapath (via ODP)` (when it decides to use `fastdp`).  However, there are no discernible error messages before or after this log message.

**Comment by bboreham** [comment]
Thanks for the report. 

It will fall back to pcap if it thinks the machine lacks vxlan support. After this decision is made and the bridge devices are configured one way or the other it will stick that way until a reboot. 

Can you try setting `WEAVE_DEBUG` to nonblank on first install (as an environment setting for the plugin)?  That should give much more log info.

**Comment by brb** [comment]
This should fix your issue: https://github.com/weaveworks/weave/pull/3080

**Comment by brb** [comment]
@dtshepherd the fixed version of the plugin was submitted to Docker Store, but it has not been acknowledged by Docker yet (usually, it takes quite some time).

Meanwhile, you could install the fixed plugin from Docker Hub by running `docker plugin install weaveworks/net-plugin:2.0.4`

**Comment by brb** [comment]
Fixed by #3080
